use-cases/meeting-scheduling/src/main/java/org/acme/meetingschedule/domain/TimeGrain.java

```diff
@@ -0,0 +1,134 @@
+package org.acme.meetingschedule.domain;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.Objects;
+
+import ai.timefold.solver.core.api.domain.lookup.PlanningId;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+@JsonIdentityInfo(scope = TimeGrain.class, generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+public class TimeGrain implements Comparable<TimeGrain> {
+
+    private static final Comparator<TimeGrain> COMPARATOR = Comparator.comparing(TimeGrain::getDayOfYear)
+            .thenComparingInt(TimeGrain::getStartingMinuteOfDay);
```

Comment:
Why is `TimeGrain` comparable and can we do anything to stop it?

Reply:
We can remove it by using a `LocalDateTime`.

Reply:
Hmm, if you say we need it for joiners in constraints, then I'm fine with `TimeGrain` being comparable. I was just making sure we're not carrying over some old cruft.

Reply:
Let me undo the change.